src/lib/Syntax.hs

```diff
@@ -71,6 +72,10 @@ data Atom = Var Var
           | Pi  PiType
           | DataCon DataDef [Atom] Int [Atom]
           | TypeCon DataDef [Atom]
+          | Record (LabeledItems Atom)
+          | RecordTy (LabeledItems Type)
+          | Variant (LabeledItems Type) Label Int Atom
```

Comment:
I wonder how is it possible that you were able to get away without something a'la `SumAsProd`? Are there any tests where the chosen variant depends on some run-time information?

Reply:
Maybe you noticed this later, but I use the existing `SumAsProd` as-is. I think some of my tests exercise that functionality when they loop over a table containing variants.